osquery/tables/system/linux/pci_devices.cpp

```diff
@@ -272,11 +410,12 @@ QueryData genPCIDevices(QueryContext& context) {
   struct udev_list_entry *device_entries, *entry;
   device_entries = udev_enumerate_get_list_entry(enumerate.get());
 
-  auto del_udev_device = [](udev_device* d) { udev_device_unref(d); };
   udev_list_entry_foreach(entry, device_entries) {
     const char* path = udev_list_entry_get_name(entry);
-    std::unique_ptr<udev_device, decltype(del_udev_device)> device(
-        udev_device_new_from_syspath(udev_handle.get(), path), del_udev_device);
+
+    std::unique_ptr<udev_device, decltype(&udev_device_unref)> device(
```

Comment:
why do you need a `&` for the type of deleter?

Reply:
Thought it had to be because of the docs (doesn't looks like it satisfies anything but `lvalue reference to function`:
```
Type requirements
--
-Deleter must be FunctionObject or lvalue reference to a FunctionObject or lvalue reference to function, callable with an argument of type unique_ptr<T, Deleter>::pointer
```